tasks/us/census/lodes.py

```diff
@@ -468,148 +466,76 @@ def columns(self):
-            )),
-            ('jobs_firm_500_more_employees', OBSColumn(
-                type='Integer',
-                name='Jobs at firms with 500 or more Employees',
-                description='Number of jobs for workers at firms with Firm Size: 500 or more Employees',
-                weight=1,
-                aggregate='sum',
-                targets={total_jobs: 'denominator'},
-                tags=[tags['employment'], tags['commerce_economy']]
-            )),
         ])
         for colname, col in cols.items():
             col.tags.append(source)
             col.tags.append(license)
         return cols
 
 
-class DownloadWorkplaceAreaCharacteristics(Task):
-    '''
-    Download all WAC files available
-    '''
-
+class DownloadUnzipLodes(DownloadGUnzipTask):
     year = IntParameter(default=2013)
+    filetype = Parameter(default='rac')
+    state = Parameter()
+    partorsegment = Parameter(default='S000')
+    jobtype = Parameter(default='JT00')
 
-    def requires(self):
-        for state in STATES - MISSING_STATES.get(self.year, set()):
-            yield DownloadLODESFile(filetype='wac',
-                                    part_or_segment='S000', # all jobs
-                                    year=self.year,
-                                    state=state)
+    def filename_lodes(self):
+        #   [STATE]_[FILETYPE]_[PART/SEG]_[TYPE]_[YEAR].csv.gz   where
+        return '{}_{}_{}_{}_{}.csv.gz'.format(self.state, self.filetype,
+                                              self.partorsegment,
+                                              self.jobtype, self.year)
 
-    def output(self):
-        for outfile in self.input():
-            yield outfile
+    def url(self):
+        return 'http://lehd.ces.census.gov/data/lodes/LODES7/{}/{}/{}'.format(
+            self.state, self.filetype, self.filename_lodes())
+
+    def download(self):
+        shell('wget -O {output}.gz {url}'.format(output=self.output().path, url=self.url()))
 
 
-class DownloadResidenceAreaCharacteristics(Task):
+class WorkplaceAreaCharacteristicsTemp(CSV2TempTableTask):
 
     year = IntParameter(default=2013)
 
-    def requires(self):
-        for state in STATES - MISSING_STATES.get(self.year, set()):
-            yield DownloadLODESFile(filetype='rac', year=self.year,
-                                    state=state, part_or_segment='S000') # all jobs
+    FILE_EXTENSION = 'csv'
 
-    def output(self):
-        for outfile in self.input():
-            yield outfile
+    def requires(self):
+        return [DownloadUnzipLodes(file_extension=self.FILE_EXTENSION,
+                                   filetype='rac', year=self.year,
+                                   state=state, partorsegment='S000',
+                                   jobtype='JT00') for state in STATES - MISSING_STATES.get(self.year, set())]
+
+    def input_csv(self):
+        csvs = []
+        for folder in self.input():
```

Comment:
Iterate in a more pythonic way :D

Reply:
Do you mean using comprehensions? That was my first approach but with two `for`s and one `if` I thought it was way less readable (but of course it's open to discussion)

Reply:
Never mind I'm thinking in ruby and `.each` xD